From b68ce28dec18a9e790d3edcbbfcf22b0a1367100 Mon Sep 17 00:00:00 2001
From: Ian Abbott <abbotti@mev.co.uk>
Date: Mon, 8 Dec 2008 17:05:50 +0000
Subject: Staging: comedi: Add a module parameter 'comedi_autoconfig'.

From: Ian Abbott <abbotti@mev.co.uk>

Set it to 0 or 'N' to disable autoconfiguration.
It is enabled by default.

From: Ian Abbott <abbotti@mev.co.uk>
Cc: Frank Mori Hess <fmhess@users.sourceforge.ne
Cc: David Schleef <ds@schleef.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/comedi/comedi_fops.c |    3 +++
 drivers/staging/comedi/comedi_fops.h |    1 +
 drivers/staging/comedi/drivers.c     |    3 +++
 3 files changed, 7 insertions(+)

--- a/drivers/staging/comedi/comedi_fops.c
+++ b/drivers/staging/comedi/comedi_fops.c
@@ -59,6 +59,9 @@ int comedi_debug;
 module_param(comedi_debug, int, 0644);
 #endif
 
+int comedi_autoconfig = 1;
+module_param(comedi_autoconfig, bool, 0444);
+
 static DEFINE_SPINLOCK(comedi_file_info_table_lock);
 static struct comedi_device_file_info
     *comedi_file_info_table[COMEDI_NUM_MINORS];
--- a/drivers/staging/comedi/comedi_fops.h
+++ b/drivers/staging/comedi/comedi_fops.h
@@ -4,5 +4,6 @@
 
 extern struct class *comedi_class;
 extern const struct file_operations comedi_fops;
+extern int comedi_autoconfig;
 
 #endif /* _COMEDI_FOPS_H */
--- a/drivers/staging/comedi/drivers.c
+++ b/drivers/staging/comedi/drivers.c
@@ -797,6 +797,9 @@ int comedi_auto_config(struct device *ha
 	struct comedi_device_file_info *dev_file_info;
 	int retval;
 
+	if (!comedi_autoconfig)
+		return -ENODEV;
+
 	minor = comedi_alloc_board_minor(hardware_device);
 	if(minor < 0) return minor;
 	dev_set_drvdata(hardware_device, (void*)(unsigned long)minor);
